lib/BC/SleighLifter.cpp

```diff
@@ -839,10 +863,48 @@ class SleighLifter::PcodeToLLVMEmitIntoBlock : public PcodeEmit {
     }
   }
 
+
+  std::optional<llvm::Type *> GetFloatTypeOfByteSize(size_t byte_size) {
+    switch (byte_size) {
+      case 4: return llvm::Type::getFloatTy(this->context);
```

Comment:
Add support for HalfTy? or Float128Ty? Might not be needed now, but perhaps doesn't work?